state/relation_test.go

```diff
@@ -947,6 +947,7 @@ func (s *RelationSuite) TestWatchApplicationSettings(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	relation, err := s.State.AddRelation(eps...)
 	c.Assert(err, jc.ErrorIsNil)
+	s.State.StartSync()
```

Comment:
You should instead use `WaitForModelWatchersIdle(c, s.State.ModelUUID())`

The reason it fails intermittently is that the state watcher will send down an initial event, but the events that are in flight from the construction of the relation may not have fully flowed through, so you end up with an additional notification.

This is why we have the WaitForModelWatcherIdle method.

Reply:
Ok, using that instead - thanks.